ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java

```diff
@@ -234,12 +234,24 @@ void run(HiveConf conf, String jobName, Table t, Partition p, StorageDescriptor
         "especially if this message repeats.  Check that compaction is running properly.  Check for any " +
         "runaway/mis-configured process writing to ACID tables, especially using Streaming Ingest API.");
       int numMinorCompactions = parsedDeltas.size() / maxDeltasToHandle;
+      parsedDeltas.sort(AcidUtils.ParsedDeltaLight::compareTo);
+
+      int start = 0;
+      int end = maxDeltasToHandle;
+
       for (int jobSubId = 0; jobSubId < numMinorCompactions; jobSubId++) {
+        while (parsedDeltas.get(end).getMinWriteId() == parsedDeltas.get(end - 1).getMinWriteId() &&
+          parsedDeltas.get(end).getMaxWriteId() == parsedDeltas.get(end - 1).getMaxWriteId()) {
```

Comment:
Sorry I don't get this part. What does it do?

Reply:
it removes deltas for statements that went of the batch size. ( [... delta_5_5_1 ] [delta_5_5_2, ...])  - we should process multiple statements for the same writeIds range in the same batch, in this scenario delta_5_5_1 should be included in the next batch.

Reply:
with hive.compactor.delta.pct.threshold=2 and above code would it give an error ?

Reply:
actually thinking some more:: if there are multiple deltas with same min,max and different statement id then we have to make sure that value of hive.compactor.delta.pct.threshold is greater than any length of such sorted list or we have to artificially at that point increase the above value with a warning in log

Reply:
'hive.compactor.delta.pct.threshold' is just one of the conditions for auto-compaction.